lib/battle.rb

```diff
@@ -40,26 +41,59 @@ def place_computer_ships
   end
 
   def instructions
-    return "I have laid out my ships on the grid.\nYou now need to lay out your ships.\nThe Cruiser is three units long and the Submarine is two units long.\n#{@user_board.render(true)}"
+    puts "I have laid out my ships on the grid.\nYou now need to lay out your ships.\nThe Cruiser is three units long and the Submarine is two units long.\n#{@user_board.render(true)}"
+    user_ship_placement
+    #placeholder for take_turn method
   end
 
   def user_ship_placement
-    return "Enter the squares for the Cruiser (3 spaces):"
-    coordinates = gets.chomp 
-    while !@user_board.valid_placement?(@ships.first, coordinates)
-      "Those are invalid coordinates. Please try again."
-      coordinates = gets.chomp
-    end
-    @user_board.place(@ships.first, coordinates)
-    @user_health += @ships.first.health
-    @user_board.render(true)
-    return "Enter the squares for the Submarine (2 spaces):"
-    coordinates_2 = gets.chomp 
-    while !@user_board.valid_placement?(@ships.last, coordinates_2)
-      "Those are invalid coordinates. Please try again."
-      coordinates_2 = gets.chomp
+    @ships.each do |ship|
+      puts "Enter the squares for the #{ship.name} (#{ship.length} spaces):"
+      coordinates = gets.chomp.split
+      until @user_board.valid_placement?(ship, coordinates)
+        puts "Those are invalid coordinates. Please try again."
+        coordinates = gets.chomp.split
+      end
+      @user_board.place(ship, coordinates)
+      @user_health += ship.health
+      puts @user_board.render(true)
```

Comment:
Added enumerable to user_ship_placement to DRY it up (and make it dynamic if we eventually use more ships)